[workflow] Update workflow docs (#24249)
ray-project/ray: doc/source/workflows/basics.rst

Here is a single three-node DAG:

.. code-block:: python

    from typing import List
    import ray

    # Define Ray remote functions.
    @ray.remote
    def read_data(num: int):
        return [i for i in range(num)]

    @ray.remote
    def preprocessing(data: List[float]) -> List[float]:
        return [d**2 for d in data]

    @ray.remote
    def aggregate(data: List[float]) -> float:
        return sum(data)

    # Build the DAG.
    data = read_data.bind(10)
    preprocessed_data = preprocessing.bind(data)
    output = aggregate.bind(preprocessed_data)

The Ray DAG will not be executed until further actions are taken on it.

Your first workflow
-------------------

A single line is all you need to turn the previous DAG into a workflow:

.. code-block:: python

    # <follow the previous code>
    from ray import workflow

    # Create the workflow from the DAG.
    wf = workflow.create(output)
    # Execute the workflow and print the result.
    print(wf.run())

    # The workflow can also be executed asynchronously.
    # print(ray.get(output.run_async()))

Here is the workflow we created:

.. image:: basic.png
   :width: 500px
   :align: center

> Each node in the original DAG becomes a workflow step.

Reviewer: ```suggestion
Each node in the original DAG becomes a workflow task.
```

> Workflow steps behave similarly to Ray tasks. They are executed in a parallel and distributed way.

Reviewer: ```suggestion
Workflow tasks behave similarly to Ray tasks. They are executed in a parallel and distributed way.
```

> We can retrieve the results for individual workflow steps too with *named steps*. A step can be named in two ways:

Reviewer: ```suggestion
We can retrieve the results for individual workflow tasks too with *named steps*. A task can be named in two ways:
```

>  1) via ``.options(name="step_name")``

Reviewer: ```suggestion
 1) via ``.options(name="task_name")``
```

>  2) via decorator ``@workflow.options(name="step_name"``)

Reviewer: ```suggestion
 2) via decorator ``@workflow.options(name="task_name"``)
```

> Once a step is given a name, the result of the step will be retrievable via ``workflow.get_output(workflow_id, name="step_name")``. If the step with the given name hasn't been executed yet, an exception will be thrown. Here are some examples:

Reviewer: ```suggestion
Once a task is given a name, the result of the task will be retrievable via ``workflow.get_output(workflow_id, name="task_name")``. If the task with the given name hasn't been executed yet, an exception will be thrown. Here are some examples:
```

Reviewer: Also for the example below, change step -> task
